.github/workflows/doc-build.yml

```diff
@@ -79,10 +78,7 @@ jobs:
     - name: install-pkgs
       run: |
         sudo apt-get update
-        sudo apt-get install -y wget python3-pip git ninja-build graphviz lcov
-        wget --no-verbose "https://github.com/doxygen/doxygen/releases/download/Release_${DOXYGEN_VERSION//./_}/doxygen-${DOXYGEN_VERSION}.linux.bin.tar.gz"
-        sudo tar xf doxygen-${DOXYGEN_VERSION}.linux.bin.tar.gz -C /opt
-        echo "/opt/doxygen-${DOXYGEN_VERSION}/bin" >> $GITHUB_PATH
+        sudo apt-get install -y wget python3-pip git ninja-build doxygen graphviz lcov
```

Comment:
we should not install any doxygen version, at least a couple of recent versions were unusable with Zephyr, let's keep a pinned version. Also, with every update, doxyfile should be updated as well.

Reply:
I tend to agree with this, but on the other hand we are installing newest versions of various python libraries here as well. 

Fixed versions only really work if we install fixed versions of dependencies as well. 

In an ideal world we would just have a docker image to build documentation (that is available to users, so they can build the docs _exactly_ how we do it here).

Reply:
Doxygen is a _special_ case, I'd say. We have a custom theme to double-check on upgrades, compatibility with breathe, etc. See for example how latest version requires some fixes: https://github.com/zephyrproject-rtos/zephyr/pull/75944